test/neural_net/style_transfer.cxx

```diff
@@ -0,0 +1,138 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#define BOOST_TEST_MODULE
+
+#include <boost/test/unit_test.hpp>
+#include <core/util/test_macros.hpp>
+#include <iostream>
+
+#ifdef HAS_MACOS_10_15
+
+struct style_transfer_test {
```

Comment:
Why do we need a separate class for this, when you have individual calls to BOOST_AUTO_TEST_CASE below?